internal/controller/httproute_controller.go

```diff
@@ -65,9 +70,93 @@ func (r *HTTPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				},
 			),
 		).
+		Watches(&v1alpha1.BackendTrafficPolicy{},
+			handler.EnqueueRequestsFromMapFunc(r.listHTTPRoutesForBackendTrafficPolicy),
+			builder.WithPredicates(
+				predicate.Funcs{
+					GenericFunc: func(e event.GenericEvent) bool {
+						return false
+					},
+					DeleteFunc: func(e event.DeleteEvent) bool {
+						return true
+					},
+					CreateFunc: func(e event.CreateEvent) bool {
+						return true
+					},
+					UpdateFunc: func(e event.UpdateEvent) bool {
+						oldObj, ok := e.ObjectOld.(*v1alpha1.BackendTrafficPolicy)
+						newObj, ok2 := e.ObjectNew.(*v1alpha1.BackendTrafficPolicy)
+						if !ok || !ok2 {
+							return false
+						}
+						oldRefs := oldObj.Spec.TargetRefs
+						newRefs := newObj.Spec.TargetRefs
+
+						// 将旧引用转换为 Map
+						oldRefMap := make(map[string]v1alpha1.BackendPolicyTargetReferenceWithSectionName)
+						for _, ref := range oldRefs {
+							key := fmt.Sprintf("%s/%s/%s", ref.Group, ref.Kind, ref.Name)
+							oldRefMap[key] = ref
+						}
+
+						for _, ref := range newRefs {
+							key := fmt.Sprintf("%s/%s/%s", ref.Group, ref.Kind, ref.Name)
+							delete(oldRefMap, key)
+						}
+						if len(oldRefMap) > 0 {
+							targetRefs := make([]v1alpha1.BackendPolicyTargetReferenceWithSectionName, 0, len(oldRefs))
+							for _, ref := range oldRefMap {
+								targetRefs = append(targetRefs, ref)
+							}
+							dump := oldObj.DeepCopy()
+							dump.Spec.TargetRefs = targetRefs
+							r.genericEvent <- event.GenericEvent{
+								Object: dump,
+							}
+						}
+						return true
+					},
+				},
+			),
+		).
+		WatchesRawSource(
+			source.Channel(
+				r.genericEvent,
+				handler.EnqueueRequestsFromMapFunc(r.listHTTPRouteForGenericEvent),
+			),
+		).
 		Complete(r)
 }
 
+func (r *HTTPRouteReconciler) listHTTPRouteForGenericEvent(ctx context.Context, obj client.Object) []reconcile.Request {
+	requests := []reconcile.Request{}
+	switch v := obj.(type) {
+	case *v1alpha1.BackendTrafficPolicy:
+		httprouteAll := []gatewayv1.HTTPRoute{}
+		for _, ref := range v.Spec.TargetRefs {
+			httprouteList := &gatewayv1.HTTPRouteList{}
+			if err := r.List(ctx, httprouteList, client.MatchingFields{
+				indexer.ServiceIndexRef: indexer.GenIndexKey(v.GetNamespace(), string(ref.Name)),
+			}); err != nil {
+				r.Log.Error(err, "failed to list HTTPRoutes for BackendTrafficPolicy", "namespace", v.GetNamespace(), "ref", ref.Name)
+				return nil
+			}
+			httprouteAll = append(httprouteAll, httprouteList.Items...)
+		}
+		for _, hr := range httprouteAll {
+			requests = append(requests, reconcile.Request{
+				NamespacedName: client.ObjectKey{
+					Namespace: hr.Namespace,
+					Name:      hr.Name,
+				},
+			})
+		}
+	default:
+		r.Log.Error(fmt.Errorf("unexpected object type"), "failed to convert object to BackendTrafficPolicy")
+	}
+	return requests
+}
```

Comment:
是否可以直接将 refs 转换成 []reconcile.Request, 而不必关心它 target 到的 HTTPRoute
```suggestion
	requests := []reconcile.Request{}
	switch v := obj.(type) {
	case *v1alpha1.BackendTrafficPolicy:
		var namespacedNames = make(map[types.NamespacedName]struct{})
		for _, ref := range v.Spec.TargetRefs {
			namespacedName := types.NamespacedName{Namespace: obj.GetNamespace(), Name: ref.Name}
			if _, ok := namespacedNames[namespacedName]; !ok {
				namespacedNames[namespacedName] = struct{}
				requests = append(requests, reconcile.Request{
					NamespacedName: namespacedName,
				})
			}
		}
	default:
		r.Log.Error(fmt.Errorf("unexpected object type"), "failed to convert object to BackendTrafficPolicy")
	}
	return requests
}
```

Reply:
No, the target is a service, not an httproute.